tests/profiles-enabled/test_shipments_states.py

```diff
@@ -251,3 +252,37 @@ def test_shadow_state_updates(api_client, mocked_iot_api, shipment_with_device):
     shipment_with_device.save()
     call_count += 1  # Status should have been updated in the shadow
     assert call_count == mocked_iot_api.call_count
+
+
+@pytest.mark.django_db
+def test_dropoff_gtx(api_client, shipment):
+    assert shipment.delivery_act is None
+
+    # Set NFC/Asset ID
+    super_secret_nfc_id = 'package123'
+    shipment.asset_physical_id = hashlib.sha256(super_secret_nfc_id.encode()).hexdigest()
```

Comment:
Shouldn't we do something similar in `ShipmentActionRequestSerializer.validate_asset_physical_id`?

Reply:
For now I have the validation of `raw_asset_physical_id` in the `drop_off` transition function. Really it could just as easily go in the `ShipmentActionRequestSerializer`, but I am leaning towards saying that any conditions required for the state transitions to succeed should be checked within the state transition functions themselves. This helps us keep all of the state transition logic in one place.

Reply:
Got it but my main concern is that there is no logic to hash the `Shipment.asset_physical_id` property before `shipment.save()` the way it is done on line 263 unless the GTX tag will always being hashed out before passing it to our API.

Reply:
Oh - sorry, the intention is that GTX will be hashing the NFC IDs themselves and saving that hash in the `asset_physical_id` before passing it to the API. And then we only verify this hash during `drop_off`. I'm going to double check this with @djones0 though.

Reply:
Checking with GTX to verify. Will keep you guys posted.